### ISSUE TITLE
Do not enforce description for private symbols

### ISSUE THREAD

**Issue by customcommander** [eslint/eslint#15197]
### What rule do you want to change?

symbol-description

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
// The answer to everything
const Answer = Symbol(); // <- Description is not enforced

const Return = what => what === Answer ? 42 : what;

// Ask the right question…
module.exports = ask => ask === "What is the answer?" ? Return(Answer) : Return("🤷‍♂️");
```

### What does the rule currently do for this code?

The  rule enforces symbol description so I must do this:

```javascript
const Answer = Symbol('The answer to everything');
```

### What will the rule do after it's changed?

As this symbol is private to the module a comment is an equally valid description for maintainers. So the rule won't ask for the symbol to describe itself.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Currently this rule may or may not have a negative impact on code minification. I tried two code minifiers and they were unable to remove the description for a private symbol and I ended up with a bundle larger than necessary.
